AuthServ/AuthServer.cpp

```diff
@@ -604,12 +635,18 @@ void cb_downloadStart(AuthServer_Private& client)
         SEND_REPLY();
         return;
     }
-    filename = filename.replace("\\", "/");
 
-    filename = DS::Settings::AuthRoot() + filename;
-    DS::FileStream* stream = new DS::FileStream();
+    // Special case: SDL files from the server' SDL directory.
+    ST_ssize_t slashPos = filename.find_last('/');
+    if (slashPos != -1 && filename.left(slashPos).compare_i("SDL") == 0 && filename.after_last('.').compare_i("sdl") == 0) {
+        filename = DS::Settings::SdlPath() + filename.substr(slashPos);
+    } else {
+        filename = DS::Settings::AuthRoot() + filename;
+    }
+
+    std::unique_ptr<DS::Stream> stream = std::make_unique<DS::FileStream>();
```

Comment:
```suggestion
    auto stream = std::make_unique<DS::FileStream>();
```
A `unique_ptr<U>` can be converted to a `unique_ptr<T>` if `U` derives from `T`, so there's no need for the extra casting complexity...

Reply:
While this is true, later in the function, a `DS::BufferStream` can be moved to this `std::unique_ptr`. I'll do some work to remove the fiddly casting to make that more clear.